src/afl-fuzz-run.c

```diff
@@ -205,14 +205,17 @@ static void write_with_gap(afl_state_t *afl, u8 *mem, u32 len, u32 skip_at,
         new_size =
             el->afl_custom_post_process(el->data, new_mem, new_size, &new_buf);
 
-        if (unlikely(!new_buf || new_size <= 0)) {
+        if (unlikely(!new_buf)) { FATAL("Custom_post_process failed"); }
 
-          FATAL("Custom_post_process failed (ret: %lu)",
-                (long unsigned)new_size);
+        if (unlikely(new_size <= 0)) {
 
-        }
+          LIST_REMOVE_CURRENT_EL_IN_FOREACH()
```

Comment:
why remove? and why remove if the size is 0 (which is a legal case, that is what this PR should be about)